recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio" and self.options.polyfill != "boost":
+            raise ConanInvalidConfiguration("For MSVC, best to use the boost polyfill")
+
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17" if self.options.polyfill == "std" else "11")
+
+    def requirements(self):
+        self.requires("mongo-c-driver/1.17.2")
+        if self.options.polyfill == "boost":
+            self.requires("boost/1.74.0")
+
+        # Cannot model mnmlstc (not packaged, is pulled dynamically) or
+        # std::experimental (how to check availability in stdlib?) polyfill
+        # dependencies
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-r" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        self._cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        self._cmake.definitions["BUILD_VERSION"] = self.version
+        self._cmake.definitions["BSONCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.definitions["MONGOCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        if self.settings.compiler == "Visual Studio":
+            cmake_file = os.path.join(self._source_subfolder, "../CMakeLists.txt")
+            tools.replace_in_file(
+                cmake_file,
+                'add_subdirectory("source_subfolder")',
+                'add_subdirectory("source_subfolder")\nadd_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)'
+            )
+
+        self._patch_sources()
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD-PARTY-NOTICES", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        # Need to ensure mongocxx is linked before bsoncxx
+        self.cpp_info.libs = sorted(tools.collect_libs(self), reverse=True)
```

Comment:
Please set the name of the `cmake` and `pkg_config` generators (as you remove `cmake` and `pkgconfig` directories in `package`)

Reply:
How to know the correct names?
I set libmongocxx everywhere for now.

Reply:
Don't remove the `pkgconfig` and `cmake` subfolder and copy what files + targets are installed in the cmake and pkgconfig folder